develop: migration of an existing psql database does not keep the current index

related to https://github.com/Open-MSS/MSS/pull/2432

I recognized that I can't add users, operations and permissions to the migrated data. 

It looks like after the migration I have now in a dump of the sql
```
--
-- Name: changes_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.changes_id_seq', 1, false);


--
-- Name: messages_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.messages_id_seq', 1, false);


--
-- Name: operations_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.operations_id_seq', 2, true);


--
-- Name: permissions_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.permissions_id_seq', 3, true);


--
-- Name: users_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.users_id_seq', 2, true);


```

I should have

```

--
-- Name: changes_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.changes_id_seq', 9987, true);


--
-- Name: messages_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.messages_id_seq', 9887, true);


--
-- Name: operations_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.operations_id_seq', 278, true);


--
-- Name: permissions_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.permissions_id_seq', 5241, true);


--
-- Name: users_id_seq; Type: SEQUENCE SET; Schema: public; Owner: mscolab
--

SELECT pg_catalog.setval('public.users_id_seq', 93, true);

```

While trying to run the migration tests against postgres locally I also get this:
```
E   sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DuplicateObject) type "messagetype" already exists
E
E   [SQL: CREATE TYPE messagetype AS ENUM ('TEXT', 'SYSTEM_MESSAGE', 'IMAGE', 'DOCUMENT')]
E   (Background on this error at: https://sqlalche.me/e/20/f405)
```

The migrations aren't yet working for postgres because alembic didn't generate code to drop the created enum types on downgrade. We really need to get #2311 done before we can somewhat confidently say that we support postgres at all...

--- a/mslib/mscolab/server.py
+++ b/mslib/mscolab/server.py
@@ -110,6 +110,34 @@ your database MSColab will abort. Please follow the documentation for a manual d
                     )
                     target_connection.execute(stmt.values(row))
             target_connection.commit()
+            if target_engine.name == "postgresql":
+                # Fix the databases auto-increment sequences, if it is a PostgreSQL database
+                # For reference, see: https://wiki.postgresql.org/wiki/Fixing_Sequences
+                logging.info("Using a PostgreSQL database, will fix up sequences")
+                cur = target_connection.execute(sqlalchemy.text(r"""
+SELECT
+    'SELECT SETVAL(' ||
+    quote_literal(quote_ident(sequence_namespace.nspname) || '.' || quote_ident(class_sequence.relname)) ||
+    ', COALESCE(MAX(' ||quote_ident(pg_attribute.attname)|| '), 1) ) FROM ' ||
+    quote_ident(table_namespace.nspname)|| '.'||quote_ident(class_table.relname)|| ';'
+FROM pg_depend
+    INNER JOIN pg_class AS class_sequence
+        ON class_sequence.oid = pg_depend.objid
+            AND class_sequence.relkind = 'S'
+    INNER JOIN pg_class AS class_table
+        ON class_table.oid = pg_depend.refobjid
+    INNER JOIN pg_attribute
+        ON pg_attribute.attrelid = class_table.oid
+            AND pg_depend.refobjsubid = pg_attribute.attnum
+    INNER JOIN pg_namespace as table_namespace
+        ON table_namespace.oid = class_table.relnamespace
+    INNER JOIN pg_namespace AS sequence_namespace
+        ON sequence_namespace.oid = class_sequence.relnamespace
+ORDER BY sequence_namespace.nspname, class_sequence.relname;
+"""))
+                for stmt, in cur.all():
+                    target_connection.execute(sqlalchemy.text(stmt))
+                target_connection.commit()
         logging.info("Data migration finished")
 
     # Upgrade to the latest database revision

--- a/tests/_test_mscolab/test_migrations.py
+++ b/tests/_test_mscolab/test_migrations.py
@@ -129,5 +129,9 @@ def test_upgrade_from(revision, iterations, mscolab_app, tmp_path):
             del actual_data_after_downgrade["alembic_version"]  # expected data doesn't have the revision table
             # Check that after a downgrade the data is definitely the same
             assert expected_data == actual_data_after_downgrade
+
+            # Try to add a new user after the migration
+            flask_migrate.upgrade(directory=migrations_path)
+            assert mslib.mscolab.seed.add_user('test123@test456', 'test123', 'test456')
     finally:
         mscolab_settings.SQLALCHEMY_DB_URI_TO_MIGRATE_FROM = None

